### PR TITLE
perf: Don't show all teams when modifying/creating users

### PR DESCRIPTION
**Pull request by markstory**
Some users have 100s of teams, rendering them all results in poor performance and isn't great to use. Instead only render the active teams and use our typeahead pattern for adding teams. Typeahead results are fetched from the server as necessary. 

New component looks like:

screen shot 2019-03-06 at 12 57 41 pm

Fixes SEN-233

### CODE REVIEW

**Comment by lynnagara** [comment]
There's a failing invite member acceptance test

**Comment by markstory** [comment]
@lynnagara All fixed up now.

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/roleSelect.jsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ class RoleSelect extends React.Component {
     const {disabled, enforceAllowed, roleList, selectedRole} = this.props;
 
     return (
-      <Panel className="new-invite-team">
+      <Panel className="new-invite-role">
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/roleSelect.jsx]: Do we need this class here? There don't seem to be any styles associated with it

**Review by lynnagara — COMMENTED** [review]
This seems to generate a pretty strange search query

e.g. `/api/0/organizations/default/teams/?0=c&1=a&2=p&3=t&4=a&5=i&6=n?0=c&1=a&2=p&3=t&4=a&5=i&6=n`

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/roleSelect.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/roleSelect.jsx]: It isn't used. I corrected it as it was wrong, but I'm happy to take it out entirely.

Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/roleSelect.jsx]: Fixed now.

**Comment by markstory** [comment]
@lynnagara Thanks for spotting that. I made a change based on prettier suggestion and mucked up the query parameter passing.

**Review by lynnagara — APPROVED** [review]

**Pull request merged**